specs/WebAuthenticationRequested.md

```diff
@@ -0,0 +1,276 @@
+  [propget] HRESULT UserName([out, retval] LPWSTR* userName);
+  /// Set user name property
+  [propput] HRESULT UserName([in] LPCWSTR userName);
+
+  /// Password provided for authentication
+  [propget] HRESULT Password([out, retval] LPWSTR* password);
+  /// Set password property
+  [propput] HRESULT Password([in] LPCWSTR password);
+}
+
+/// Event args for the BasicAuthenticationRequested event. Will contain the
+/// request that led to the HTTP authentication challenge, the challenge
+/// and allows the host to provide credentials response or cancel the request.
+[uuid(51d3adaa-159f-4e48-ad39-a86beb2c1435), object, pointer_default(unique)]
+interface ICoreWebView2BasicAuthenticationRequestedEventArgs : IUnknown
+{
+  /// The web resource request that led to the authentication challenge
+  [propget] HRESULT Request([out, retval] ICoreWebView2WebResourceRequest** request);
+
+  /// The authentication challenge string
+  [propget] HRESULT Challenge([out, retval] LPWSTR* challenge);
+
+  /// Response to the authentication request with credentials. This object will be populated by the app
+  /// if the host would like to provide authentication credentials.
+  [propget] HRESULT Response([out, retval] ICoreWebView2BasicAuthenticationResponse** response);
+  
+  /// Cancel the authentication request. False by default.
+  /// If set to true, Response set will be ignored.
+  [propget] HRESULT Cancel([out, retval] BOOL* cancel);
+  /// Set the Cancel property.
+  [propput] HRESULT Cancel([in] BOOL cancel);
+
+  /// Returns an `ICoreWebView2Deferral` object. Use this deferral to
+  /// defer the decision to show the Basic Authentication dialog.
+  HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+}
+
+```
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    /// Event args for the BasicAuthenticationRequested event. Will contain the
+    /// request that led to the HTTP authentication challenge, the challenge
+    /// and allows the host to provide authentication response or cancel the request.
+    runtimeclass CoreWebViewBasicAuthenticationRequestedEventArgs
+    {
+        /// The web resource request that led to the authentication challenge
+        CoreWebView2WebResourceRequest Request { get; };
+
+        /// The HTTP basic authentication challenge string
+        String Challenge { get; };
+
+        /// Cancel the authentication request. False by default.
+        /// If set to true, Response will be ignored.
+        bool Cancel { get; set; };
+
+        /// Response to the authentication request with credentials. This object will be populated by the app
+        /// if the host would like to provide authentication credentials.
+        CoreWebView2BasicAuthenticationResponse Response { get; };
+
+        /// Returns an `ICoreWebView2Deferral` object. Use this deferral to
+        /// defer the decision to show the Basic Authentication dialog.
+        CoreWebView2Deferral Deferral { get; };
```

Comment:
Should be a method `GetDeferral()`, not a property. (Sample code above uses `GetDeferral()` method.)

Properties with read side effects are scary, because the debugger will try to fetch all properties when you hover over the object or expand it in the "Locals" pane. That will retrieve the Deferral property, print it, and then throw it away. The deferral is now leaked, and the act of debugging your event prevented it from working.

Reply:
Yes we use GetDeferral everywhere else in our API. This should be a method not a property.